### PR TITLE
First release of helm chart in new location

### LINKED ISSUE #3: release first version of the chart in the new location

### PR DESCRIPTION
**Pull request by torstenwalter**
This is also the first release of the jenkins chart in the new Github org. The README was updated to reflect the new location.

This also adds https://github.com/github/super-linter GitHub action to validate markdown files.
To ensure a nice contributor experience all linting errors have been fixed.

Note: yaml linting is disabled as Helm templates are no valid yaml.

Closes #3

### CODE REVIEW

File: CODE_OF_CONDUCT.md
Comment (scottrigby) [inline CODE_OF_CONDUCT.md:1]: along with this, delete https://github.com/jenkinsci/helm-charts/blob/main/code-of-conduct.md

File: charts/jenkins/README.md
Comment (scottrigby) [inline charts/jenkins/README.md:28]: In general, adopt the README simplified template we added for the new prometheus charts. Example: https://github.com/scottrigby/prometheus-helm-charts/tree/main/charts/prometheus

Comment (scottrigby) [reply charts/jenkins/README.md:28]: I made a PR against this PR branch in your fork. https://github.com/torstenwalter/helm-charts/pull/1

File: CODE_OF_CONDUCT.md
Comment (torstenwalter) [reply CODE_OF_CONDUCT.md:1]: good catch.

File: README.md
Comment (timja) [inline README.md:15]: we can probably setup a CNAME if that would be preferred?

https://charts.jenkins.io ?

e.g. send a PR here:
https://github.com/jenkins-infra/azure/pull/154/files

Comment (torstenwalter) [reply README.md:15]: Do you have a name in mind? I've been thinking of `charts.jenkins.io`. Using that the URL would be `helm repo add jenkinsci https://charts.jenkins.io/helm-charts`.

Comment (timja) [reply README.md:15]: that was the name i had in mine 😄

Comment (torstenwalter) [reply README.md:15]: Created a PR for it: https://github.com/jenkins-infra/azure/pull/156

Comment (torstenwalter) [reply README.md]: It's been changed to helm repo add jenkinsci https://charts.jenkins.io

**Comment by oleg-nenashev** [comment]
CC @jenkinsci/code-reviewers

File: charts/jenkins/CHANGELOG.md
Comment (oleg-nenashev) [inline charts/jenkins/CHANGELOG.md:8]: Should this file get migrated to the repository root? if not, it might make sense to reference it from README

File: charts/jenkins/README.md
Comment (oleg-nenashev) [inline charts/jenkins/README.md:121]: These tables might have had some value as user documentation for those who have no immediate access to Helm, but I'd guess maintaining them is difficult

**Review by oleg-nenashev — APPROVED** [review]
Looks good overall! Note that the "Jenkins master" terminology is now deprecated, it would be great to cleanup the terminology in subsequent pull requests

File: charts/jenkins/CHANGELOG.md
Comment (torstenwalter) [reply charts/jenkins/CHANGELOG.md:13]: It\s now referenced from the chart README.

File: charts/jenkins/README.md
Comment (torstenwalter) [reply charts/jenkins/README.md:121]: @wmcdona89 added them back in a separate file.

**Comment by scottrigby** [comment]
Curious why values summary was added back. Generally these tend to get out of sync with values.yaml, which hopefully is already readable-ideally more so.

**Comment by torstenwalter** [comment]
https://github.com/torstenwalter/helm-charts/pull/3#discussion_r479716337

**Comment by torstenwalter** [comment]
@wmcdona89 I am not really sure if it's a good idea to keep the values table. It's quite some effort to keep it up to date. When removing values or just change the default value of a key one always has to update the table. I think values.yaml should be authorities. We could link to that file in the README instead.

**Comment by scottrigby** [comment]
It's really up to you all as maintainers of the chart. In my PR to your branch I just applied the same template as proposed for the prometheus charts, for the same reasons: 1. simplifying (charts have become a bit overcomplicated, the repeated README table convention is just one way), 2. easier maintenance (and support for when users get frustrated when table values don't work as expected, since they're manually copied for each chart.values change and periodically are wrong), and 3. the more clear and simple chart Readmes are overall the better for end users. My opinion though, just wanted to ask if returning those was intentional 😄

**Comment by torstenwalter** [comment]
I added some more changes:
- top-level README now at least contains a reference to the Jenkins chart
- migration steps from one version to another is moved into a migration guide section at the bottom. I think it's something people are only interested once, while the more common use case is to look up configuration examples.
- moved the JCasC configuration to the top of the configuration section as this one is an enabler for users as it allows to make the majority of changes.

I also noticed that we still have XML configuration examples in the README. These should be removed or replaces with JCasC alternatives in future PRs.

**Comment by torstenwalter** [comment]
I had another look in the file describing the values. What I really like about it is the grouping of related values together. Therefor I am ok with leaving them for now.

Generating it from `values.yaml` will not provide us the same experience. Using new keys for values to do the grouping will be a breaking change. What we could try is to sort them in the `value.yaml` the same way and use comments for seperation. Downside is that keys are not sorted alphabetically.

**Comment by torstenwalter** [comment]
So this PR is up for a last review round. We should try to move future improvements to new PRs to unblock the migration and allow people to contribute.

**Comment by wmcdona89** [comment]
I opened issue https://github.com/jenkinsci/helm-charts/issues/18 to organize `values.yaml` to align with the sections in `VALUES_SUMMARY`. then the `README` can link to each section and `VALUES_SUMMARY` can be removed.

**Pull request merged**
